File: 프로그래머스/lv1/42862. 체육복/체육복.py

def solution(n, lost, reserve):
    lost_copy = [x for x in lost if x not in reserve]
    reserve_copy = [x for x in reserve if x not in lost]
    lost_copy.sort()
    answer = n - len(lost_copy)
    for each in lost_copy:
        if each - 1 in reserve_copy:
          reserve_copy.remove(each - 1)
          answer += 1
        elif each + 1 in reserve_copy:
          reserve_copy.remove(each + 1)
          answer += 1
    return answer

def main():
  print(solution(3, [3], [1]))

if __name__ == "__main__":
    main()

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    lost.sort((a, b) => a - b);
    reserve.sort((a, b) => a - b);
    let realLost = lost.filter((l) => !reserve.includes(l));
    const realReserve = reserve.filter((re) => !lost.includes(re));
    for (res of realReserve) {
        if (realLost.includes(res - 1)) {
            realLost = realLost.filter((l) => l !== res - 1);
        } else if (realLost.includes(res + 1)) {
            realLost = realLost.filter((l) => l !== res + 1)
        }
    }
    return n - realLost.length;
}